### PR TITLE
[Experiment] Support experiment run anonymous exp

### PR DESCRIPTION
**Pull request by lalala123123**
# Description
Python code:
```python
experiment_path = EXP_ROOT / "basic-script-template" / "basic-script.exp.yaml"
experiment = _load_experiment(experiment_path)
client = PFClient()
exp = client._experiments.start(experiment, inputs={"key": "value"}, stream=True)
```

CLI:
```
# Start a named experiment:
pf experiment start -n my_experiment --inputs data1=data1_val data2=data2_val --stream
# Run an experiment by yaml file:
pf experiment start --file path/to/my_experiment.exp.yaml --inputs data1=data1_val data2=data2_val --stream
```

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_sdk/operations/_experiment_operations.py [hunk 1]
```diff
@@ -121,6 +122,24 @@ def stop(self, name: str, **kwargs) -> Experiment:
         ExperimentOrchestrator(self._client, self.get(name)).stop()
         return self.get(name)
 
+    @monitor_operation(activity_name="pf.experiment.run", activity_type=ActivityType.PUBLICAPI)
+    def run(self, experiment: Experiment, stream=False, **kwargs) -> Experiment:
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/operations/_experiment_operations.py:126]: let's consider the cli experience:

pf experiment start --file xx.exp.yaml
pf experiment start --name xxx

We should also have stop?
pf experiment stop --file xx.exp.yaml
pf experiment stop --name xx?

Or we name this like 
pf experiment stop --experiment xx.exp.yaml/name

File: src/promptflow/promptflow/_cli/_pf/_experiment.py [hunk 2]
```diff
@@ -135,6 +159,24 @@ def add_experiment_stop(subparsers):
     )
 
 
+def add_experiment_run(subparsers):
+    epilog = """
+    Examples:
+
+    # Run an experiment by yaml file:
+    pf experiment run --file path/to/my_experiment.exp.yaml --inputs data1=data1_val data2=data2_val --stream
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_cli/_pf/_experiment.py:167]: let's do it via start

File: src/promptflow/promptflow/_sdk/operations/_experiment_operations.py (hunk 1 above)
Comment (lalala123123) [reply src/promptflow/promptflow/_sdk/operations/_experiment_operations.py]: offline sync:
pf experiment start --name exp_name --> pfclient.experiment.start(experiment=exp_object)
pf experiment start --file path/to/exp.yaml --inputs key=value --stream --> pfclinet.experiment.run(experiment=exp_obj, inputs={}, stream=False)
 
pf experiment stop --name exp_name --> pfclient.experiment.stop(experiment=exp_object)

File: src/promptflow/promptflow/_cli/_pf/_experiment.py (hunk 2 above)
Comment (lalala123123) [reply src/promptflow/promptflow/_cli/_pf/_experiment.py]: remove pf experiment run

**Review by wangchao1230 — DISMISSED** [review]

File: src/promptflow/promptflow/_cli/_pf/_experiment.py [hunk 3]
```diff
@@ -197,11 +226,41 @@ def show_experiment(args: argparse.Namespace):
...
     print(json.dumps(result._to_dict(), indent=4))
 
 
 def stop_experiment(args: argparse.Namespace):
-    result = _get_pf_client()._experiments.stop(args.name)
+    client = _get_pf_client()
+    if args.name:
+        logger.debug(f"Stop a named experiment {args.name}.")
+        experiment = client._experiments.get(args.name)
+    elif args.file:
```
Comment (0mza987) [inline src/promptflow/promptflow/_cli/_pf/_experiment.py:260]: So anonymous exp run does not have run record? This is a bit strange to stop an exp with its yaml file. I thought we will record the anonymous exp run but just give it a random exp name guid.

Comment (brynn-code) [reply src/promptflow/promptflow/_cli/_pf/_experiment.py:260]: +1 what's the target scenario of passing file here?

Comment (lalala123123) [reply src/promptflow/promptflow/_cli/_pf/_experiment.py:260]: This is a new added user experience in Evaluation Experience.
It is used to support users to directly execute from experience yaml. And it also supports future feature that users can define experiences through Python code.

**Review by wangchao1230 — APPROVED** [review]

**Review by 0mza987 — APPROVED** [review]

**Pull request merged**